eth/contracts/TreasureHuntCreator.sol

```diff
@@ -71,4 +71,11 @@ contract TreasureHuntCreator is Ownable {
   function getAddressHash(address a) pure public returns (bytes32) {
     return keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n20", a));
   }
+
+  function getLeaderboard(uint offset) view public returns(uint[64] memory leaderboard) {
+    for(uint i = 0; i < 64 && i + offset < _players.length; i++) {
+      address player = _players[i+offset];
+      leaderboard[i] = uint256(player) << 96 | uint256(_playerToCurrentChapter[player]);
+    }
+  }
```

Comment:
could you briefly explain the reasoning behind the offset logic?

Reply:
The function returns a `uint[64]` fixed-length array, where **64** is an arbitrary number.

If we have 2 players, calling `getLeaderboard(0)` will return:
```
[
  player-0-and-chapter,
  player-1-and-chapter,
  0x0,
  ...0x0 60 times...,
  0x0
]
```

If we have 80 players, then we need pagination. The first call `getLeaderboard(0)` will return: `[player-0-and-chapter, ..., player-63-and-chapter]`
The second call will have an offset of `64` to skip the players we already got, so calling `getLeaderBoard(64)` will return:
```
[
  player-64-and-chapter,
  player-65-and-chapter,
 ...,
  player-79-and-chapter,
  0x0,
  ...0x0 46 times...,
  0x0
]
```

Maybe `offset` should be renamed to `page`, so you ask for page 0, 1, 2, and more, so you don't need to know the size of a page (64 in our case).

Reply:
nice solution 👍 can you put the 64 (which is used both in the return type and the loop) into a constant like `PAGE_SIZE`? It would make things clearer i think